modconfig/flowpipe_credential.go

```diff
@@ -2122,6 +2122,49 @@ func (c *GuardrailsCredential) Validate() hcl.Diagnostics {
 	return hcl.Diagnostics{}
 }
 
+type MastodonCredential struct {
+	HclResourceImpl
+	ResourceWithMetadataImpl
+
+	Type string `json:"type" cty:"type" hcl:"type,label"`
+
+	Server      *string `json:"server,omitempty" cty:"server" hcl:"server,optional"`
+	AccessToken *string `json:"access_token,omitempty" cty:"access_token" hcl:"access_token,optional"`
+}
+
+func (*MastodonCredential) GetCredentialType() string {
+	return "mastodon"
+}
+
+func (c *MastodonCredential) getEnv() map[string]cty.Value {
+	env := map[string]cty.Value{}
+	return env
+}
+
+func (c *MastodonCredential) CtyValue() (cty.Value, error) {
+	ctyValue, err := GetCtyValue(c)
+	if err != nil {
+		return cty.NilVal, err
+	}
+
+	valueMap := ctyValue.AsValueMap()
+	valueMap["env"] = cty.ObjectVal(c.getEnv())
+
+	return cty.ObjectVal(valueMap), nil
+}
+
+func (c *MastodonCredential) Resolve(ctx context.Context) (Credential, error) {
```

Comment:
@karanpopat there's no env variable for this?

Reply:
I checked the steampipe plugin, it does not use any env variables and neither does a third-party [terraform provider](https://registry.terraform.io/providers/feditools/mastodon/latest/docs). Also mastodon has no standard env variables mentioned anywhere in docs.

Is it mandatory to have env vars?